server/service/src/repack/generate.rs

```diff
@@ -0,0 +1,204 @@
+    pub stock_lines: Vec<StockLineRow>,
+    pub location_movement: Option<Vec<LocationMovementRow>>,
+}
+
+struct StockLineJob {
+    new_stock_line: StockLineRow,
+    stock_line_to_update: StockLineRow,
+}
+
+pub fn generate(
+    ctx: &ServiceContext,
+    stock_line: StockLineRow,
+    input: InsertRepack,
+) -> Result<GenerateRepack, RepositoryError> {
+    let StockLineJob {
+        stock_line_to_update,
+        new_stock_line,
+    } = generate_new_stock_lines(&stock_line, &input);
+
+    let (repack_invoice, repack_invoice_lines) =
+        generate_invoice_and_lines(&ctx, &stock_line, &new_stock_line)?;
+    let location_movement = if let Some(_) = input.new_location_id {
+        Some(generate_location_movement(
+            &ctx.store_id,
+            &stock_line,
+            &new_stock_line,
+        ))
+    } else {
+        None
+    };
+
+    let stock_lines = vec![stock_line_to_update, new_stock_line.clone()];
+
+    Ok(GenerateRepack {
+        repack_invoice,
+        repack_invoice_lines,
+        stock_lines,
+        location_movement,
+    })
+}
+
+fn generate_invoice_and_lines(
+    ctx: &ServiceContext,
+    stock_line: &StockLineRow,
+    new_stock_line: &StockLineRow,
+) -> Result<(InvoiceRow, Vec<InvoiceLineRow>), RepositoryError> {
+    let connection = &ctx.connection;
+
+    let repack_name = NameRowRepository::new(connection)
+        .find_one_by_code("repack")?
+        .ok_or(RepositoryError::NotFound)?;
+
+    let invoice = InvoiceRow {
+        id: uuid(),
+        name_id: repack_name.id,
+        store_id: ctx.store_id.clone(),
+        user_id: Some(ctx.user_id.clone()),
+        invoice_number: next_number(connection, &NumberRowType::Repack, &ctx.store_id)?,
+        r#type: InvoiceRowType::Repack,
+        status: InvoiceRowStatus::Verified,
+        on_hold: false,
+        created_datetime: Utc::now().naive_utc(),
+        verified_datetime: Some(Utc::now().naive_utc()),
+        ..Default::default()
+    };
+
+    let mut invoice_lines = Vec::new();
+
+    let item = ItemRowRepository::new(connection)
+        .find_one_by_id(&stock_line.item_id)?
+        .ok_or(RepositoryError::NotFound)?;
+
+    let stock_in = InvoiceLineRow {
+        id: uuid(),
+        invoice_id: invoice.id.clone(),
+        item_id: stock_line.item_id.clone(),
+        item_name: item.name.clone(),
+        item_code: item.code.clone(),
+        stock_line_id: Some(new_stock_line.id.clone()),
+        location_id: new_stock_line.location_id.clone(),
+        batch: stock_line.batch.clone(),
+        expiry_date: stock_line.expiry_date,
+        pack_size: new_stock_line.pack_size,
+        r#type: InvoiceLineRowType::StockIn,
+        number_of_packs: new_stock_line.total_number_of_packs,
+        cost_price_per_pack: new_stock_line.cost_price_per_pack,
+        sell_price_per_pack: new_stock_line.sell_price_per_pack,
+        ..Default::default()
+    };
+
+    let stock_out = inline_edit(&stock_in, |mut l| {
```

Comment:
I personally have been moving away form inline_init and inline_edit, in favour of spread `..stock_in.clone()`